core/src/stored/stored_conf.cc

```diff
@@ -171,6 +171,8 @@ static ResourceItem dev_items[] = {
   {"RequiresMount", CFG_TYPE_BIT, ITEM(res_dev, cap_bits), CAP_REQMOUNT, CFG_ITEM_DEFAULT, "off", NULL, NULL},
   {"OfflineOnUnmount", CFG_TYPE_BIT, ITEM(res_dev, cap_bits), CAP_OFFLINEUNMOUNT, CFG_ITEM_DEFAULT, "off", NULL, NULL},
   {"BlockChecksum", CFG_TYPE_BIT, ITEM(res_dev, cap_bits), CAP_BLOCKCHECKSUM, CFG_ITEM_DEFAULT, "on", NULL, NULL},
+  {"ReadOnly", CFG_TYPE_BOOL, ITEM(res_dev, read_only), 0, CFG_ITEM_DEFAULT, "false", NULL, NULL},
+  {"WriteOnly", CFG_TYPE_BOOL, ITEM(res_dev, write_only), 0, CFG_ITEM_DEFAULT, "false", NULL, NULL},
```

Comment:
I totally understand that you didn't want to add a new type to the configuration parser, but right now with the two booleans we can have four states:
1. `(ReadOnly == false && WriteOnly == false)` the old default behaviour
2. `(ReadOnly == true && WriteOnly == false)` only use this device for reading
3. `(ReadOnly == false && WriteOnly == true)` only use this device for writing
4. `(ReadOnly == true && WriteOnly == true)` what does this mean?

So I would argue that we should use one parameter of type [IO_DIRECTION](https://docs.bareos.org/Configuration/CustomizingTheConfiguration.html#datatype-io_direction) to configure this behaviour.
However, I have no thorough idea how to actually name that parameter.

In the process it would probably a good idea to refactor IO_DIRECTION to use "read" and "write" instead of "in" and "out" while retaining backwards compatibility.

Reply:
You're right, that makes much more sense. I'll look into it.
How about calling it "Access Mode"?

Reply:
That sounds promising! Maybe then IO_DIRECTION should accept "readonly", "writeonly" and "readwrite", so we would end up with something like:
```
Access Mode = readonly
```
in the configuration.

Reply:
I've made it use IO_DIRECTION now and changed IO_DIRECTION to also accept read(only)/write(only) and readwrite. Next I'd rename AutoXflateMode and its values since it's no longer specific to the AutoXflate plugin. Am I on the right track with this?

Reply:
Yes, it's definitely a good idea to rename that.

Also, please note that you should rebase your branch onto the latest master.